crates/trigger/src/runtime_config/key_value.rs

```diff
@@ -122,19 +124,33 @@ impl RedisKeyValueStoreOpts {
 
 #[derive(Clone, Debug, Deserialize)]
 pub struct AzureCosmosConfig {
+    #[serde(default)]
     key: String,
     account: String,
     database: String,
     container: String,
+    #[serde(default)]
+    environmental_auth: bool,
```

Comment:
If this is set to `true`, my assumption would be that the runtime would look for the default Azure environment variables and use them.

For this to be usable in SpinKube, ahead of the Spin version update we need to update the runtime config types in the operator to support it — https://github.com/spinkube/spin-operator/blob/main/internal/runtimeconfig/runtime_config.go

Reply:
I'm not sure that is the case for Workflow Identity since those env vars are set by the Workflow Identity mutating web hook for the pod specs with the appropriate label. SpinKube would need to be able to set the labels and the service account for the pod spec, so it would work with Workload Identity. 

For other auth types (say service principal w/ secret), that would likely need to have env vars specified on the SpinApp to then be translated to pod spec env vars.

Reply:
I've removed the `environmental_auth` field and am defaulting to env auth when `key` is not provided.